splunk-otel-android/src/main/java/com/splunk/rum/ActivityTracer.java

```diff
@@ -0,0 +1,143 @@
+        return this;
+    }
+
+    ActivityTracer startActivityCreation() {
+        spanAndScope.startSpan(this::makeCreationSpan);
+        return this;
+    }
+
+    private Span makeCreationSpan() {
+        //If the application has never loaded an activity, or this is the initial activity getting re-created,
+        // we name this span specially to show that it's the application starting up. Otherwise, use
+        // the activity class name as the base of the span name.
+        boolean isColdStart = initialAppActivity.get() == null;
+        if (isColdStart) {
+            return createSpanWithParent("Created", appStartupTimer.getStartupSpan());
+        }
+        if (activityName.equals(initialAppActivity.get())) {
+            return createAppStartSpan("warm");
+        }
+        return createSpan("Created");
+    }
+
+    ActivityTracer initiateRestartSpanIfNecessary(boolean multiActivityApp) {
+        if (spanAndScope.spanInProgress()) {
+            return this;
+        }
+        spanAndScope.startSpan(() -> makeRestartSpan(multiActivityApp));
+        return this;
+    }
+
+    @NonNull
+    private Span makeRestartSpan(boolean multiActivityApp) {
+        //restarting the first activity is a "hot" AppStart
+        //Note: in a multi-activity application, navigating back to the first activity can trigger
+        //this, so it would not be ideal to call it an AppStart.
+        if (!multiActivityApp && activityName.equals(initialAppActivity.get())) {
+            return createAppStartSpan("hot");
+        }
+        return createSpan("Restarted");
+    }
+
+    private Span createAppStartSpan(String startType) {
+        Span span = createSpan(APP_START_SPAN_NAME);
+        span.setAttribute(SplunkRum.START_TYPE_KEY, startType);
+        //override the component to be appstart
+        span.setAttribute(SplunkRum.COMPONENT_KEY, SplunkRum.COMPONENT_APPSTART);
+        return span;
+    }
+
+    private Span createSpan(String spanName) {
+        return createSpanWithParent(spanName, null);
+    }
+
+    private Span createSpanWithParent(String spanName, Span parentSpan) {
+        final SpanBuilder spanBuilder = tracer.spanBuilder(spanName)
+                .setAttribute(ACTIVITY_NAME_KEY, activityName)
+                .setAttribute(SplunkRum.COMPONENT_KEY, SplunkRum.COMPONENT_UI);
+        if (parentSpan != null) {
+            spanBuilder.setParent(parentSpan.storeInContext(Context.current()));
+        }
+        Span span = spanBuilder.startSpan();
+        //do this after the span is started, so we can override the default screen.name set by the RumAttributeAppender.
+        span.setAttribute(SplunkRum.SCREEN_NAME_KEY, activityName);
+        return span;
+    }
+
+    void endSpanForActivityResumed() {
+        if (initialAppActivity.get() == null) {
+            initialAppActivity.set(activityName);
+        }
+        endActiveSpan();
+    }
+
+    void endActiveSpan() {
+        if (appStartupTimer != null) {
```

Comment:
`appStartupTimer` can't really be null anymore - that used to be possible when you used the `NamedTrackableTracer` for fragment tracing

Reply:
good catch. I thought I had found all these, but I missed that one.